src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayOfStates = [];
+  const stateCopy = Object.assign({}, state);
+  const addPropertiesAction = 'addProperties';
+  const removePropertiesAction = 'removeProperties';
+  const clearAction = 'clear';
+
+  for (const action of actions) {
+    const { type, extraData = null, keysToRemove = null } = { ...action };
```

Comment:
```suggestion
    const { 
      type, 
      extraData = null, 
      keysToRemove = null,
    } = { ...action };
```
If you set the default values, you should check if these values are present before using them